### PR TITLE
api/discord: add DefaultMemberPermissions and NoDMPermission to commands

### PR DESCRIPTION
**Pull request by Juby210**

### CODE REVIEW

File: discord/command.go [hunk 1]
```diff
@@ -52,6 +52,11 @@ type Command struct {
 	//
 	// It is only present on ChatInputCommands.
 	Options CommandOptions `json:"options,omitempty"`
+	// DefaultMemberPermissions is set of permissions.
+	DefaultMemberPermissions Permissions `json:"default_member_permissions,string,omitempty"`
+	// DmPermission indicates whether the command is available in DMs with
+	// the app, only for globally-scoped commands. By default, commands are visible.
+	DmPermission bool `json:"dm_permission,omitempty"`
```
Comment (topi314) [inline discord/command.go]: This fields defaults to true
Perhaps it should be inverted like `NoDefaultPermission`

File: discord/command.go [hunk 2]
```diff
@@ -52,6 +52,11 @@ type Command struct {
 	//
 	// It is only present on ChatInputCommands.
 	Options CommandOptions `json:"options,omitempty"`
+	// DefaultMemberPermissions is set of permissions.
+	DefaultMemberPermissions Permissions `json:"default_member_permissions,string,omitempty"`
```
Comment (topi314) [inline discord/command.go]: 0 is a valid permission here you can set to disallow anyone except it they have admin or an override to use the command, if it's 0 it gets omitted perhaps thats not ideal

Comment (Juby210) [reply discord/command.go]: I see, do you have any proposition to fix this?

Comment (topi314) [reply discord/command.go]: I guess a pointer would do the job, not sure if it is the best solution here
how does arikawa does it for other fields like this?

File: api/application.go [hunk 3]
```diff
@@ -26,6 +26,8 @@ type CreateCommandData struct {
 	Description              string                 `json:"description"`
 	DescriptionLocalizations discord.StringLocales  `json:"description_localizations,omitempty"`
 	Options                  discord.CommandOptions `json:"options,omitempty"`
+	DefaultMemberPermissions discord.Permissions    `json:"default_member_permissions,string,omitempty"`
+	NoDmPermission           bool                   `json:"-"`
```
Comment (diamondburned) [inline api/application.go]: This should be `NoDMPermission`.

File: api/application.go [hunk 4]
```diff
@@ -34,13 +36,15 @@ func (c CreateCommandData) MarshalJSON() ([]byte, error) {
 	type RawCreateCommandData CreateCommandData
 	cmd := struct {
 		RawCreateCommandData
+		DmPermission      bool `json:"dm_permission"`
```
Comment (diamondburned) [inline api/application.go]: `DMPermission`

File: discord/command.go (hunk 2 above)
Comment (diamondburned) [reply discord/command.go]: You can use `option.Uint`, but `*Permissions` can work, too. I would make a helper function `NewPermissions(p ...Permissions) *Permissions` in that case.

**Comment by diamondburned** [comment]
Above changes must be done before this PR can be merged.

**Comment by diamondburned** [comment]
Also, please link the documentation for these 2 fields in the code.

**Comment by Juby210** [comment]
> Also, please link the documentation for these 2 fields in the code.

Documentation is linked to struct itself, so I guess this is enough.
Also applied your suggestions.

**Review by diamondburned — APPROVED** [review]

**Pull request merged**